db/plain_table_db_test.cc

```diff
@@ -770,8 +770,19 @@ TEST_P(PlainTableDBTest, BloomSchema) {
     for (unsigned i = 0; i < 32; ++i) {
       // Known pattern of Bloom filter false positives can detect schema change
       // with high probability. Known FPs stuffed into bits:
-      bool expect_fp = (bloom_locality ? 2421694657UL : 1785868347UL)
-                       & (1UL << i);
+      uint32_t pattern;
+      if (!bloom_locality) {
+        pattern = 1785868347UL;
+      } else if (CACHE_LINE_SIZE == 64) {
+        pattern = 2421694657UL;
+      } else if (CACHE_LINE_SIZE == 128) {
+        pattern = 788710956UL;
+      } else {
+        ASSERT_EQ(CACHE_LINE_SIZE, 256);
```

Comment:
Now appveyor fails with:
```
:\projects\rocksdb\third-party\gtest-1.8.1\fused-src\gtest\gtest.h(21207): error C2220: warning treated as error - no 'object' file generated [C:\projects\rocksdb\build\rocksdb_plain_table_db_test.vcxproj]
         c:\projects\rocksdb\third-party\gtest-1.8.1\fused-src\gtest\gtest.h(21235): note: see reference to function template instantiation 'testing::AssertionResult testing::internal::CmpHelperEQ<T1,T2>(const char *,const char *,const T1 &,const T2 &)' being compiled
                 with
                 [
                     T1=unsigned int,
                     T2=int
   300>Link:
            Creating library C:/projects/rocksdb/build/Debug/persistent_cache_test.lib and object C:/projects/rocksdb/build/Debug/persistent_cache_test.exp
   301>ClCompile:
                 ]
         c:\projects\rocksdb\db\plain_table_db_test.cc(781): note: see reference to function template instantiation 'testing::AssertionResult testing::internal::EqHelper<false>::Compare<unsigned int,int>(const char *,const char *,const T1 &,const T2 &)' being compiled
                 with
                 [
                     T1=unsigned int,
                     T2=int
                 ]
         c:\projects\rocksdb\db\plain_table_db_test.cc(781): note: see reference to function template instantiation 'testing::AssertionResult testing::internal::EqHelper<false>::Compare<unsigned int,int>(const char *,const char *,const T1 &,const T2 &)' being compiled
                 with
                 [
                     T1=unsigned int,
                     T2=int
                 ]
   301>c:\projects\rocksdb\third-party\gtest-1.8.1\fused-src\gtest\gtest.h(21207): warning C4389: '==': signed/unsigned mismatch [C:\projects\rocksdb\build\rocksdb_plain_table_db_test.vcxproj]
```
I guess there was a data race between a Windows build option was checked and this one. So I think need to fix this assertion.

Reply:
Anand caught this first: https://github.com/facebook/rocksdb/pull/5788